src/frontend/src/lib/derived/address.derived.ts

```diff
@@ -9,24 +9,49 @@ export const addressNotLoaded: Readable<boolean> = derived([addressStore], ([$ad
 	isNullish($addressStore)
 );
 
-export const btcAddressMainnet: Readable<OptionBtcAddress> = derived(
+const btcAddressMainnetData: Readable<AddressData | null | undefined> = derived(
 	[addressStore],
-	([$addressStore]) =>
-		$addressStore?.[BTC_MAINNET_TOKEN_ID] === null
-			? null
-			: $addressStore?.[BTC_MAINNET_TOKEN_ID]?.data
+	([$addressStore]) => $addressStore?.[BTC_MAINNET_TOKEN_ID]
 );
 
-export const ethAddress: Readable<OptionEthAddress> = derived([addressStore], ([$addressStore]) =>
-	$addressStore?.[ETHEREUM_TOKEN_ID] === null ? null : $addressStore?.[ETHEREUM_TOKEN_ID]?.data
+const ethAddressData: Readable<AddressData | null | undefined> = derived(
+	[addressStore],
+	([$addressStore]) => $addressStore?.[ETHEREUM_TOKEN_ID]
 );
 
-export const addressCertified: Readable<boolean> = derived(
-	[addressStore],
-	([$addressStore]) => $addressStore?.[ETHEREUM_TOKEN_ID]?.certified === true
+export const btcAddressMainnet: Readable<OptionBtcAddress> = derived(
+	[btcAddressMainnetData],
+	([$btcAddressMainnetData]) =>
+		$btcAddressMainnetData === null ? null : $btcAddressMainnetData?.data
+);
+
+export const ethAddress: Readable<OptionEthAddress> = derived(
+	[ethAddressData],
+	([$ethAddressData]) => ($ethAddressData === null ? null : $ethAddressData?.data)
+);
+
+const btcAddressMainnetCertified: Readable<boolean> = derived(
+	[btcAddressMainnetData],
+	([$btcAddressMainnetData]) => $btcAddressMainnetData?.certified === true
+);
+
+export const btcAddressMainnetNotCertified: Readable<boolean> = derived(
+	[btcAddressMainnetCertified],
+	([$btcAddressMainnetCertified]) => !$btcAddressMainnetCertified
+);
+
+const ethAddressCertified: Readable<boolean> = derived(
```

Comment:
Do we need global derived stores? According current implementation aren't we only checking for certification in a single component? If we do need one, can you provide the refactoring in a separate PR?

Reply:
not really needed to be honest... I just wanted to keep it more immediately clear, than just having the notCertified store. But it makes sense to reduce the stores. Correcting